src/compiler.py

```diff
@@ -961,7 +951,10 @@ def compile_bql(self, bdb, bql, out):
         assert ast.is_bql(bql)
         generator_id = self.generator_id
         if isinstance(bql, ast.ExpBQLProb):
-            raise BQLError(bdb, 'Probability of value is 1-row function.')
+            compile_pdf_joint(bdb, generator_id, self.modelno, bql.targets,
+                bql.constraints, self, out)
+        elif isinstance(bql, ast.ExpBQLProbFn):
+            raise BQLError(bdb, 'Probability of value is 1-column function.')
```

Comment:
I tweaked a few more of them.  But that's getting afield from the issue at hand.  Feel free to update the error messages in another branch, or open an issue for ones you find confusing.